### ISSUE TITLE
Re-enable upload of coverage files to codecov

### ISSUE THREAD

**Issue by richard-cox** [rancher/dashboard#11106]
- in the `test` workflow we use the action `codecov/codecov-action@v3` to upload unit test coverage to codecov servers
- the action has been approved for use, but with caveats (it must be in it's on workflow)
  - see security issue 728 (request @richard-cox for details)
- given timeline the action will be removed from `test` but the functionality needs to come back
- we can either
  - create a new workflow triggered on `test` workflow, download artefacts (these are already uploaded) and then upload to codecov
  - OR
  - use the codecov cli to upload from within `test`

**Comment by rak-phillip** [comment]
It appears that rke2 is actively using codecov[^1]. We might be able to learn from their existing approach

[^1]: https://github.com/rancher/rke2/tree/master/.github

**Comment by cnotv** [comment]
For the records:
- Company level usage https://app.codecov.io/gh/rancher/
- Current version is `codecov/codecov-action@v4`

**Comment by torchiaf** [comment]
Blocked by https://github.com/rancherlabs/eio/issues/2484

**Comment by gaktive** [comment]
Looks like this card was archived when it was reopened. Putting this in 2.11.0 for now though @torchiaf is this still blocked on EIO since that ticket is closed?

**Comment by torchiaf** [comment]
@gaktive we can go ahead with this now that EIO migrated the secrets. It's fine to move in 2.11.0.

**Comment by nwmac** [comment]
@torchiaf What is the status of this? Its for 2.14 and re-opened? What do we need to do?

**Comment by nwmac** [comment]
Closing as won't fix
